buildSrc/src/main/java/xyz/wagyourtail/doclet/tsdoclet/PackageTree.java

```diff
@@ -55,16 +56,41 @@ private void prepareTSTree() {
         while (this.dirty) {
             this.dirty = false;
             for (ClassParser aClass : Set.copyOf(classes)) {
+                redirects.add(aClass.getClassName(false));
+            }
+            for (ClassParser aClass : Set.copyOf(classes)) {
+                aClass.redirects.addAll(redirects);
                 compiledClasses.computeIfAbsent(aClass, ClassParser::genTSInterface);
             }
             for (PackageTree value : Set.copyOf(children.values())) {
+                value.redirects.addAll(redirects);
                 value.prepareTSTree();
             }
         }
     }
 
+    /**
+     * this exist because the method above seems like will miss some type
+     * idk why to use Set.copyOf, i'm just afraid of breaking something so i'd use it as well
```

Comment:
Set.copyOf is because while resolving classes they can point to other classes that need to be resolved, and so this can cause a concurrentModificationException, instead we use the original list, and just mark it as dirty so it starts over again... 

I suppose when it becomes dirty we could immediately jump back to the top of the while loop to make it a little faster tho.

Reply:
i don't even know why would it become dirty again, maybe i'm blind

Reply:
i tried it anyway, doesn't seem like faster or slower